### ISSUE TITLE
Support real and imag at the backend level

### ISSUE THREAD

**Issue by alewis** [google/TensorNetwork#841]

**Comment by ahmedkrmn** [comment]
Hi @alewis,
Any ideas on how to tackle this?

**Comment by MikeMerz** [comment]
Hey,
Is this issue still still open?

**Comment by alewis** [comment]
Yep, this one is, though I don't know whether @ahmedkrmn is working on it?

**Comment by ahmedkrmn** [comment]
@MikeMerz, feel free to work on it if you want to.
I've been bus lately so I still haven't' started.

**Comment by MikeMerz** [comment]
@ahmedkrmn Thanks mate.
Just a though:
1)For numpy and tensorflow backends, it's possible to just to call numpy and tensor's .real and .imag? as the input is of the same type.

**Comment by alewis** [comment]
Yes, that should work in all cases except PyTorch and Symmetric. For
PyTorch you'll need to do some additional type checking because calling .
real and .imag throws an error for non complex input in that case.

**Comment by MikeMerz** [comment]
For PyTorch, is it better to raise an excpetion if the input is not complex or treat it as 0j?

**Comment by alewis** [comment]
I think the latter, yeah.

**Comment by MikeMerz** [comment]
@alewis Hey, I unintentionally pushed the changes of this Issue to the ones_like and zero_like Pull request

**Comment by alewis** [comment]
Hmm, do you think you can revert it and split them up?

**Comment by MikeMerz** [comment]
@alewis after much tries, I re-uploaded the changes for the ones_like & zero_like issue , had to close the last request

**Comment by AnuravModak** [comment]
Heyy @MikeMerz , @alewis  I am a new contributor can i take up thi s?? and can someone provide little heads-up or description about this issue??
